docs/API.md

```diff
@@ -0,0 +1,176 @@
+# Shipyard API
+
+Logically, the API has three parts to handle the three areas of functionality in Shipyard.
+
+1. Document Staging
+2. Action Handling 
+3. Airflow Monitoring 
+
+There are several standard error responses that should be handled appropriately by a client of this API:
+* 401 Authentication Error  
+If not authenticated
+* 403 Forbidden  
+If not authorized
+* 404 Not found  
+If a variable in the path doesn't reference a valid resource 
+* 503 Service Unavailable  
+Error if an upstream system cannot be reached (e.g. Deckhand for documents, Airflow for actions)
+
+---
```

Comment:
Added a section for headers used across the APIs, if there are some that end up being specific to the resource/method, I'll add them in there.